jupyterlab_gcedetails/src/components/select_input.tsx

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as React from 'react';
+import { stylesheet } from 'typestyle';
+import ArrowDropDownIcon from '@material-ui/icons/ArrowDropDown';
+import TextField from '@material-ui/core/TextField';
+import MenuItem from '@material-ui/core/MenuItem';
+import { Option, TEXT_STYLE } from '../data';
+
+interface Props {
+  label?: string;
+  name?: string;
+  value?: string;
+  options?: Option[];
+  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
+}
+
+export const STYLES = stylesheet({
+  select: {
+    display: 'block',
+    marginTop: '4px',
+  },
+  icon: {
+    right: '14px',
+    position: 'absolute',
+  },
+});
+
+const iconComponent = props => {
+  return <ArrowDropDownIcon className={STYLES.icon} />;
+};
+
+export function SelectInput(props: Props) {
+  const { label, name, value, options, onChange } = props;
+
+  return (
+    <div>
+      {label && <label>{label}</label>}
+      <TextField
```

Comment:
Just curious, what's the advantage of the TextField vs. the Select React material component?

Reply:
I'm specifically using it to get the styling of the machine type nested select component that I built and this select input to match better. I originally tried using the Select React material component but some of their default styling made this harder to do. It may also make sense to extract the textfield out into a wrapper component and have the machine type nested select component and the select input component use that instead so styling across all future inputs for this extension are consistent but I think for now this is fine, what do you think?

Reply:
SGTM